src/ddsrt/src/ifaddrs/windows/ifaddrs.c

```diff
@@ -189,10 +189,14 @@ copyaddr(
   if ((ifa = ddsrt_calloc_s(1, sizeof(*ifa))) == NULL) {
     err = DDS_RETCODE_OUT_OF_RESOURCES;
   } else {
+    size_t namelen = wcslen(iface->FriendlyName);
     ifa->flags = getflags(iface);
     ifa->type = guess_iftype(iface);
     ifa->addr = ddsrt_memdup(sa, sz);
-    (void)ddsrt_asprintf(&ifa->name, "%wS", iface->FriendlyName);
+    ifa->name = ddsrt_malloc(namelen + 1);
+    if (ifa->name != NULL) {
+      wcstombs(ifa->name, iface->FriendlyName, namelen);
```

Comment:
`wcstombs` is locale dependent itself. Most of the time it'll probably be alright, but I think it would be better to use `_mbstowcs_s_l` and force the encoding to be UTF-8. `_mbstowcs_s_l` can also be used to determine the number of bytes required in the destination judging by the [documentation](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mbstowcs-s-mbstowcs-s-l).